autoload/vital/__vital__/System/File.vim

```
@@ -312,22 +329,19 @@ elseif s:is_unix
 
 elseif s:is_windows
   function! s:rmdir(path, ...) abort
+    let path = a:path
     let flags = a:0 ? a:1 : ''
-    if &shell =~? 'sh$'
-      let cmd = flags =~# 'r' ? 'rm -rf' : 'rmdir'
-      let ret = system(cmd . ' ' . shellescape(a:path))
-    else
-      " 'f' flag does not make sense.
-      let cmd = 'rmdir /Q'
-      let cmd .= flags =~# 'r' ? ' /S' : ''
-      let ret = system(cmd . ' "' . a:path . '"')
-    endif
-    if v:shell_error
-      let ret = iconv(ret, 'char', &encoding)
-      throw 'vital: System.File: rmdir(): ' . substitute(ret, '\n', '', 'g')
+    " All characters must be encoded to system encoding.
+    if s:need_trans
+      let path  = iconv(path, &encoding, 'char')
     endif
-  endfunction
 
+    " 'f' flag does not make sense.
+    let cmd = 'rmdir /Q'
+    let cmd .= flags =~# 'r' ? ' /S' : ''
+    call system(cmd . ' "' . path . '"')
+    return !v:shell_error
+  endfunction
```

In order to keep old behavior, it should throw an exception with command output?

```vim
    let ret = system(cmd . ' "' . path . '"')
    if v:shell_error
      throw 'vital: System.File: rmdir(): ' . substitute(ret, '\n', '', 'g')
    endif
```